plugins/inputs/t128_metrics/t128_metrics_test.go

```diff
@@ -0,0 +1,512 @@
+package t128_metrics_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"net/http/httptest"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	plugin "github.com/influxdata/telegraf/plugins/inputs/t128_metrics"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+type Endpoint struct {
+	URL             string
+	Code            int
+	ExpectedRequest string
+	Response        string
+}
+
+var ResponseProcessingTestCases = []struct {
+	Name              string
+	ConfiguredMetrics []plugin.ConfiguredMetric
+	Endpoints         []Endpoint
+	ExpectedMetrics   []*testutil.Metric
+	ExpectedErrors    []string
+}{
+	{
+		Name: "empty results produce no metrics",
+		ConfiguredMetrics: []plugin.ConfiguredMetric{{
+			"test-metric",
+			map[string]string{"test-field": "stats/test"},
+			map[string][]string{},
+		}},
+		Endpoints:       []Endpoint{{"/stats/test", 200, "{}", "[]"}},
+		ExpectedMetrics: nil,
+		ExpectedErrors:  nil,
+	},
+	{
+		Name: "none value produces no metric",
+		ConfiguredMetrics: []plugin.ConfiguredMetric{{
+			"test-metric",
+			map[string]string{"test-field": "stats/test"},
+			map[string][]string{},
+		}},
+		Endpoints: []Endpoint{{"/stats/test", 200, "{}", `[{
+			"id": "/stats/test-metric",
+			"permutations": [{
+				"parameters": [],
+				"value": null
+			}]
+		}]`}},
+		ExpectedMetrics: nil,
+		ExpectedErrors:  nil,
+	},
+	{
+		Name: "forms string value if it is non numeric",
+		ConfiguredMetrics: []plugin.ConfiguredMetric{{
```

Comment:
is there a tests to validate empty ConfiguredMetrics?

Reply:
Nope, I'll add one